dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AppController.java

```diff
@@ -164,66 +163,85 @@ public void reloadApps() {
   @GetMapping("/{app}/**")
   public void renderApp(
       @PathVariable("app") String appName, HttpServletRequest request, HttpServletResponse response)
-      throws IOException, WebMessageException, ForbiddenException {
+      throws IOException, WebMessageException {
     String contextPath = request.getContextPath();
     String baseUrl = contextService.getContextPath();
     App application = appManager.getApp(appName, baseUrl);
 
     if (application == null) {
+      log.info("App not found: {}", appName);
       throw new WebMessageException(notFound("App '" + appName + "' not found."));
     }
 
     if (!appManager.isAccessible(application)) {
-      throw new ForbiddenException("You don't have access to application " + appName + ".");
+      log.info("User does not have access to app: {}", appName);
+      throw new WebMessageException(
+          forbidden("User does not have access to app '" + appName + "'."));
     }
```

Comment:
getApp above already checks `isAccessible`, so this code path will technically always be skipped and a 403 never returned.  We get a 404 if the user doesn't have access to the app.  I left this here for now in case we want to move it up and go back to returning a 403.

Note that this was an existing bug for a long time, so for *installed apps* we have never returned 403.  Bundled apps were handled with hard-coded authority checks based on their path, so they did return top-level 403 errors.  Since we've harmonized how bundled and installed apps are served we need to pick one behavior - I've kept the 404 response for now to avoid disclosing the presence of apps to unauthorized users.